grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -858,39 +858,62 @@ public void validVertex(Vertex vertex){
+     * @return true if the given set of Ids contains concepts that are duplicates of the main concept
      */
-    @Override
-    public boolean fixDuplicateCastings(String index, Set<ConceptId> castingVertexIds){
-        Set<CastingImpl> duplicated = castingVertexIds.stream()
-                .map(id -> this.<CastingImpl>getConceptRawId(id.getValue()))
+    private Set<? extends ConceptImpl> getDuplicates(ConceptImpl mainConcept, Set<ConceptId> conceptVertexIds){
+        Set<ConceptImpl> duplicated = conceptVertexIds.stream()
+                .map(id -> this.<ConceptImpl>getConceptRawId(id.getValue()))
                 //filter non-null, will be null if previously deleted/merged
                 .filter(Objects::nonNull)
                 .collect(toSet());
 
+        duplicated.remove(mainConcept);
+
+        return duplicated;
+    }
+
+    /**
+     * Given an index, get the concept associated with it
+     * @param index retrieve the concept associated with this index
+     * @return Concept representing the vertex at the given index
+     */
+    private ConceptImpl getMainConcept(String index){
         //This is done to ensure we merge into the indexed casting. Needs to be cleaned up though
-        CastingImpl mainCasting = getConcept(Schema.ConceptProperty.INDEX, index);
-        duplicated.remove(mainCasting);
+        return getConcept(Schema.ConceptProperty.INDEX, index);
+    }
 
-        if(duplicated.size() > 0){
-            //Fix the duplicates
-            Set<Relation> duplicateRelations = mergeCastings(mainCasting, duplicated);
+    /**
+     * Merges the provided duplicate castings.
+     *
+     * @param castingVertexIds The vertex Ids of the duplicate castings
+     * @return if castings were merged, a commit is required and the casting index exists
+     */
+    @Override
+    public boolean fixDuplicateCastings(String index, Set<ConceptId> castingVertexIds){
+        CastingImpl mainCasting = (CastingImpl) getMainConcept(index);
+        Set<CastingImpl> duplicated = (Set<CastingImpl>) getDuplicates(mainCasting, castingVertexIds);
 
-            //Remove Redundant Relations
-            duplicateRelations.forEach(relation -> ((ConceptImpl) relation).deleteNode());
+        //Fix the duplicates
+        Set<Relation> duplicateRelations = mergeCastings(mainCasting, duplicated);
 
-            //Restore the index
-            String newIndex = mainCasting.getIndex();
-            mainCasting.getVertex().property(Schema.ConceptProperty.INDEX.name(), newIndex);
+        //Remove Redundant Relations
+        duplicateRelations.forEach(relation -> ((ConceptImpl) relation).deleteNode());
 
-            return true;
-        }
+        //Restore the index
+        String newIndex = mainCasting.getIndex();
+        mainCasting.getVertex().property(Schema.ConceptProperty.INDEX.name(), newIndex);
 
-        return false;
+        return true;
```

Comment:
Yeah, 100% agreed. Especially if the return value means that something has been modified and you should commit.